src/Type/Definition/UnionType.php

```diff
@@ -57,7 +58,7 @@ public function isPossibleType(Type $type) : bool
         if ($this->possibleTypeNames === null) {
             $this->possibleTypeNames = [];
             foreach ($this->getTypes() as $possibleType) {
-                $this->possibleTypeNames[$possibleType->name] = true;
+                $this->possibleTypeNames[Type::resolveLazyType($possibleType)->name] = true;
```

Comment:
We don't need it. `getTypes` always returns resolved types

Reply:
Right again! 🔥